src/react/connect.ts

```diff
@@ -65,3 +67,38 @@ export const connect =
 
     return React.createElement(Component, newProps);
   };
+
+let instanceCount = 0;
+const nextInstanceName = () => `view-${instanceCount++}`;
+
+/**
+ * Higher order component for connecting a component to a stream
+ *
+ * This is the same as {@link connect }, but takes a factory for the stream. A
+ * new stream will be retrieved from the factory for every new mount of the
+ * component.
+ *
+ * @param Component Component that will receive props from the stream
+ * @param createInstanceStream Function to create the stream for each instance
+ *  of the component
+ * @returns A component that renders the original component with props retrieved
+ *  from the stream
+ * @see {@link connect}
+ */
+export const connectInstance =
+  <Props extends Record<string, unknown>, Observed extends Partial<Props>>(
+    Component: ComponentType<Props>,
+    createInstanceStream: (instance: string) => ObservableInput<Observed>
+  ) =>
+  (props: Omit<Props, keyof Observed>) => {
+    const [stream$] = useState(() => createInstanceStream(nextInstanceName()));
+    const value = useStream(stream$);
+
+    if (value === NOT_YET_EMITTED) return null;
+
+    // Typescript doesn't recognize this as Observed & Props for some reason
+    // Question on StackOverflow: https://stackoverflow.com/q/60758084/1104307
+    const newProps = { ...props, ...value } as Props & Observed;
```

Comment:
Does it think `value` still might be `NOT_YET_EMITTED`? Does casting `value` to `Observed` help?

Reply:
BTW, I would not use the same name for the type and the constant in the case of `NOT_YET_EMITTED`...